v2/apiserver/internal/core/jobs.go

```diff
@@ -643,6 +649,31 @@ func (j *jobsService) Cleanup(
 	return nil
 }
 
+func (j *jobsService) Timeout(
+	ctx context.Context,
+	eventID string,
+	jobName string,
+) error {
+	if err := j.authorize(ctx, RoleObserver, ""); err != nil {
+		return err
+	}
+
+	// Update job status
+	status := JobStatus{
+		Phase: JobPhaseTimedOut,
+	}
+	if err := j.UpdateStatus(ctx, eventID, jobName, status); err != nil {
```

Comment:
Is there other information in the job's status that we're losing here? iirc, there's a start and end time. Do we want to preserve the start time and add in the end time?